sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -113,6 +113,42 @@ joint_matrix_apply(Group sg, joint_matrix<Group, T, Use, M, N, Layout> &jm,
   return;
 }
 
+template <typename Group, typename T, use Use, size_t M, size_t N,
+          layout Layout, typename F>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_apply(Group sg, joint_matrix<Group, T, Use, M, N, Layout> &jmsrc,
+                   joint_matrix<Group, T, Use, M, N, Layout> &jmdest,
+                   F &&lambda) {
+#if defined(__SYCL_DEVICE_ONLY__)
+#if defined(__NVPTX__) || defined(__HIP_PLATFORM_AMD_MFMA__)
+  std::ignore = sg;
+  for (int i = 0; i < jmsrc.matrix_impl.wi_marray.size(); i++) {
+    lambda(jmsrc.matrix_impl.wi_marray[i], jmdest.matrix_impl.wi_marray[i]);
+  }
+#else // NVPTX
+  using storage_element_type =
+      typename oneapi::detail::jm_type_interpretation_helper_trait<
+          T>::storage_element_type;
+  auto wi_data_c = sycl::ext::oneapi::detail::get_wi_data(sg, jmsrc);
+  auto wi_data_d = sycl::ext::oneapi::detail::get_wi_data(sg, jmdest);
+  for (int i = 0; i < wi_data_c.length(); i++) {
+    storage_element_type elementsrc = wi_data_c[i];
+    storage_element_type elementdest = wi_data_d[i];
+    lambda(elementsrc, elementdest);
+    wi_data_d[i] = elementdest;
```

Comment:
What if `elementsrc` is also modified in lambda function. Should we also add `wi_data_c[i] = elementsrc;` to cover case like that?
If we prohibit changing of `elementsrc` in that case, should we add some guards for that in spec? like `const` modifier or pass by value, not reference?

Reply:
You are right that the intention of this new overload is to only modify the destination matrix. I will add more clarification to the spec. 
I can add const modifier in the examples to emphasize that.